lib/ctx.cpp

```diff
@@ -2762,6 +2762,11 @@ grn_ctx_output_result_set_open(grn_ctx *ctx,
                              result_set,
                              format,
                              n_additional_elements);
+  grn_output_result_set_content(ctx,
+                                ctx->impl->output.buf,
+                                ctx->impl->output.type,
+                                result_set,
+                                format);
```

Comment:
これは最終的にはここに入れないで独立したAPIにしたいんだけど、それは別PR？

普通の人は`grn_ctx_output_result_set()`を使って、こだわる人は個別に`grn_ctx_output_result_set_{open,content,close}()`を呼ぶようにしたい。

Reply:
後方互換な話で `grn_ctx_output_result_set_open()` の挙動が変わるのは良くないかと思ったんですが、気にしなくて良いですか？

Reply:
誰も使っていないだろうからいいんじゃね？と思っていたけど、ケアしちゃう？

だったら、

```diff
diff --git a/include/groonga/output.h b/include/groonga/output.h
index ee6370da7..c107e1cb3 100644
--- a/include/groonga/output.h
+++ b/include/groonga/output.h
@@ -38,6 +38,7 @@ typedef struct _grn_obj_format grn_obj_format;
    The "1024" value may be changed.
    Since 10.0.3 */
 #define GRN_OBJ_FORMAT_AUTO_FLUSH (0x01 << 6)
+#define GRN_OBJ_FORMAT_STREAM_RECORDS (0x01 << 7)
 
 struct _grn_obj_format {
   grn_obj columns;
```

で、このフラグがないときは既存の挙動（`open`でレコードも出力して閉じちゃう）で、フラグがあるときは`open`で`records:`だけして、`content`でレコードを出力して`close`で`records`を閉じる、かなぁ。

既存の挙動だと`records: []`のあとに違う要素（ドリルダウン結果とか）を追加できて、今やろうとしているやつだと、`records:`の中身を複数回に分けて出力できるけど、違う要素は追加できないんだよなぁ。

そもそも`records:`の出力をオプションでやめようよが https://github.com/groonga/groonga/pull/2031 かぁ。

うーん、どうするのがいいんだろう。。。

Reply:
うーん、今やってもらおうとしているやつ（`records`を何回も出力できるけど他の要素を追加できない）が一番微妙かも。。。

Reply:
やるなら`result_set_open()`で`records:`も出力しないようにするべきだよなぁ。あるいは既存のように`records: [...]`を全部出力しちゃうか。

ちょっとだけはダメだな。

Reply:
`grn_output_result_set_open()`の挙動はこのままで、こんな感じになる`grn_output_result_set_open_header()`を導入する？

```c
grn_output_result_set_open()
{
  grn_output_result_set_open_header();
  grn_output_table_records();
}

Reply:
そうですね、そうしましょう。
`records` は `grn_output_table_records();` みたいに切り出されてたほうがきれいな気がします。